src/ServiceControl/MessageFailures/Api/LockedHeaderModificationValidator.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace ServiceControl.MessageFailures.Api
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public class LockedHeaderModificationValidator
+    {
+        public bool Check(string[] lockedHeaders, IList<KeyValuePair<string, string>> editedMessageHeaders, Dictionary<string, string> originalMessageHeaders)
+        {
+            foreach (var header in lockedHeaders)
+            {
+                var originalHeader = originalMessageHeaders.FirstOrDefault(x => string.Compare(x.Key, header, StringComparison.InvariantCulture) == 0);
```

Comment:
Why can't the dictionary be a one that enforces this equality comparer so TryGetValue can be used? Below too, why does it have to come in as an IList of KeyValuePairs?